File: awesomeimagepicker/src/main/java/in/myinnos/awesomeimagepicker/models/Video.java
package in.myinnos.awesomeimagepicker.models;

import android.net.Uri;
import android.os.Build;
import android.os.Parcel;

/**
 * Created by MyInnos on 03-11-2016.
 */
public class Video extends Media {

    private long duration;

    public Video() {

    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(getId());
        dest.writeString(getName());
        dest.writeLong(getDuration());
        dest.writeLong(getSize());
        dest.writeString(getMimeType());
        dest.writeString(getUri().toString());
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            dest.writeBoolean(isFavorite());
        }
    }

    public static final Creator<Video> CREATOR = new Creator<Video>() {
        @Override
        public Video createFromParcel(Parcel source) {
            return new Video(source);
        }

        @Override
        public Video[] newArray(int size) {
            return new Video[size];
        }
    };

    private Video(Parcel in) {
        setId(in.readLong());
        setName(in.readString());
        setDuration(in.readLong());
        setSize(in.readLong());
        setMimeType(in.readString());
        setUri(Uri.parse(in.readString()));
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            setFavorite(in.readBoolean());
        }
    }

    public long getDuration() {
        return duration;
    }

    public void setDuration(long duration) {
        this.duration = duration;
    }
}

File: awesomeimagepicker/src/main/java/in/myinnos/awesomeimagepicker/models/Video.kt
package `in`.myinnos.awesomeimagepicker.models

import android.net.Uri
import android.os.Build
import android.os.Parcel
import android.os.Parcelable

class Video : Media {

    var duration: Long = 0

    constructor()

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(id)
        dest.writeString(name)
        dest.writeLong(duration)
        dest.writeString(uri.toString())
        dest.writeLong(size)
        dest.writeString(mimeType)
        dest.writeLong(albumId)
        dest.writeString(albumName)
        dest.writeLong(dateAddedSecond)
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            dest.writeBoolean(isFavorite)
        }
    }

    private constructor(parcel: Parcel) {
        id = parcel.readLong()
        name = parcel.readString()
        duration = parcel.readLong()
        uri = Uri.parse(parcel.readString())
        size = parcel.readLong()
        mimeType = parcel.readString()
        albumId = parcel.readLong()
        albumName = parcel.readString()
        dateAddedSecond = parcel.readLong()
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            isFavorite = parcel.readBoolean()
        }
    }

    companion object CREATOR : Parcelable.Creator<Video> {
        override fun createFromParcel(parcel: Parcel): Video {
            return Video(parcel)
        }

        override fun newArray(size: Int): Array<Video?> {
            return arrayOfNulls(size)
        }
    }
}
